src/Functions/LeastGreatestGeneric.h

```diff
@@ -53,27 +53,39 @@ class FunctionLeastGreatestGeneric : public IFunction
         if (1 == num_arguments)
             return arguments[0].column;
 
-        Columns converted_columns(num_arguments);
+        Columns converted_columns;
+        /// Remove the only null column, not need to compare
         for (size_t arg = 0; arg < num_arguments; ++arg)
-            converted_columns[arg] = castColumn(arguments[arg], result_type)->convertToFullColumnIfConst();
+        {
+            if (!arguments[arg].type->onlyNull())
+            {
+                auto converted_col = castColumn(arguments[arg], result_type)->convertToFullColumnIfConst();
+                converted_columns.emplace_back(converted_col);
+            }
+        }
+
+        if (converted_columns.empty())
+            return arguments[0].column;
+        else if (converted_columns.size() == 1)
+            return converted_columns[0];
 
         auto result_column = result_type->createColumn();
         result_column->reserve(input_rows_count);
 
         for (size_t row_num = 0; row_num < input_rows_count; ++row_num)
         {
             size_t best_arg = 0;
-            for (size_t arg = 1; arg < num_arguments; ++arg)
+            for (size_t arg = 1; arg < converted_columns.size(); ++arg)
             {
-                auto cmp_result = converted_columns[arg]->compareAt(row_num, row_num, *converted_columns[best_arg], 1);
```

Comment:
Since you duplicated l. 68 into calls with last argument = `1` and `-1` ... are you saying that the code was _incorrect_  before?

Reply:
Yes, it is incorrect to always set the `nan_direction_hint` to be 1 at the `compareAt` function when encounter the null input, as we can see at the functions comments
```
 * If one of element's value is NaN or NULLs, then:
    * - if nan_direction_hint == -1, NaN and NULLs are considered as least than everything other;
    * - if nan_direction_hint ==  1, NaN and NULLs are considered as greatest than everything other.
```
So it we always set the value as 1, the NULL input  would be consider as greatest value, and will lead to wrong result of `greatest` function.